client/layout/guided-tours/positioning.js

```diff
@@ -82,21 +82,26 @@ export const posToCss = ( { x, y } ) => ( {
 	left: x ? x + 'px' : undefined,
 } );
 
+function hasNonEmptyClientRect( el ) {
+	const rect = el.getBoundingClientRect();
+	return ! ( rect.x === 0 && rect.y === 0 && rect.width === 0 && rect.height === 0 );
+}
+
+// discern between tip targets and regular CSS by grepping for CSS-only characters
+const CSS_SELECTOR_REGEX = /[.# ]/;
+const isCssSelector = targetSlug => CSS_SELECTOR_REGEX.test( targetSlug );
+
 export function targetForSlug( targetSlug ) {
 	if ( ! targetSlug ) {
 		return null;
 	}
-	if (
-		targetSlug.indexOf( '.' ) !== -1 ||
-		targetSlug.indexOf( '#' ) !== -1 ||
-		targetSlug.indexOf( ' ' ) !== -1
-	) {
-		// a sort of hacky way to discern tip targets and regular css for now
-		// (e.g. misses #ids, ...)
-		// TODO(lsinger): fix this
-		return query( targetSlug )[ 0 ];
-	}
-	return query( '[data-tip-target="' + targetSlug + '"]' )[ 0 ];
+
+	const cssSelector = isCssSelector( targetSlug )
+		? targetSlug
+		: `[data-tip-target="${ targetSlug }"]`;
+
+	const targetEls = query( cssSelector );
+	return find( targetEls, hasNonEmptyClientRect ) || null;
```

Comment:
why the default to `null` here instead of letting `find` return `undefined`?

Reply:
> why the default to null here instead of letting find return undefined?

Good point, it doesn't really matter. Either it returns element or a falsy value. There's a preexisting statement in this function that does
```js
if ( ! targetSlug ) {
  return null;
}
```
I just wanted to keep the return type `element | null` rather than `element | null | undefined`. Theoretically, some strict typechecker might start to complain about the latter.

Reply:
ah, cool. I missed the early return of `null`.